### ISSUE TITLE
Rename Job Subscription to Job Worker

### ISSUE THREAD

**Issue by ThorbenLindhauer** [zeebe-io/zeebe#802]
To distinguish between a topic subscription and a job subscription the later should be renamed to job worker. This should express that the worker not only handles events but also generates new commands for on the topic, i.e. task completion/failure.

The client builder API could be adjust to something along the lines:

```
TopicSubscription sub = client.newTopicSubscription().handler(RecordHandler handler).subscribe();

JobWorkerRegistration reg = client.newJobWorker().worker(JobWorker worker).register();
```

We have to check if this is a protocol change, i.e. if protocol types, command or event fields have to be adjusted.

<hr/>
Original issue description:

During https://github.com/zeebe-io/zeebe/issues/793 we noticed that people sometimes confuse the types of subscriptions and that it is a little cumbersome to explain the difference when both are called "subscriptions".

This is a proposal to find clearly distinct names for these two things. 

Why?

* They implement very different use cases; naming them such that they are easy to distinguish will make it easier to remember/recognize/understand the distinction
* The fact that a job subscription is a kind of subscription is rather a technical detail than relevant to the user (however, it does convey the meaning that this is something ongoing and that jobs are received and handled as they appear)
* Many people think of pub-sub when they hear "subscription". Exclusive locking and handling does not fit that.

How could we name it (this is the hard part)?

* `TopicSubscription` and `JobSubscription` => `RecordSubscription` and `JobWorker`
* ..
* your idea!

**Comment by Zelldon** [comment]
Maybe we can also rename locking an job to assign an job?

**Comment by ThorbenLindhauer** [comment]
Let's have a separate issue for this. My spontaneous thoughts on this are as follows: 

The "locking" concept has three concerns. 1) assignment to a worker 2) exclusiveness 3) expiration. Ideally, I would like to have a term that conveys all three. "locking" covers 2, "assignment" covers 1 and (somewhat) 2.

**Comment by ThorbenLindhauer** [comment]
To explain the differences of the subscriptions at bpmNext, Ryan called `TaskSubscriptions` *doers* and `TopicSubscriptions` *viewers*. This in more formal terms would be a nice naming in my opinion.

**Comment by Zelldon** [comment]
@ThorbenLindhauer aggree, created a separate issue for this https://github.com/zeebe-io/zeebe/issues/819

**Comment by ThorbenLindhauer** [comment]
@berndruecker any thoughts on this?

**Comment by berndruecker** [comment]
I back this oppinion because `EventHandler` is very common in different communities - so `JobHandler` should be intutive.

**Comment by menski** [comment]
Decision: TopicSubscription and JobWorker

maybe API like this:
```
TopicSubscription sub = client.newTopicSubscription().handler(RecordHandler handler).subscribe();
JobWorkerRegistration reg = client.newJobWorker().worker(JobWorker worker).register();
```

**Comment by ThorbenLindhauer** [comment]
Context for the decision: we went with `JobWorker` instead of `JobHandler` because this concept is typically described as workers. For example, when you look at a Zeebe architecture from a high level, you have brokers and job workers and that's how we explain it to users in presentations and documentation. We would not use *handler* in this context. So we would like to name it consistently in the API.

**Comment by menski** [comment]
I update the issue title and description to reflect our decision.

**Comment by ThorbenLindhauer** [comment]
Implementing this we gave this some more thought and will build the following API:

```
JobClient jobClient = ..;
JobHandler paymentHandler = ..;

JobWorker worker = jobClient
    .newWorker()
    .jobType("payment")
    .handler(paymentHandler)
    .open();
```

So we have both `JobHandler` and `JobWorker`, where the handler is the business logic to perform for each job and the worker is the stateful thing that is connected to the Zeebe cluster and can be closed etc. We believe that this is better aligned with the documented high-level and intuitive worker concept (i.e. I set up a Zeebe cluster and I connect workers to the cluster; the above code does exactly that, it creates and connects a new worker).
